mono/metadata/icall.c

```diff
@@ -3576,7 +3586,7 @@ read_enum_value (const char *mem, int type)
 }
 
 static void
-write_enum_value (char *mem, int type, guint64 value)
+write_enum_value (void *mem, int type, guint64 value)
```

Comment:
Safer way to  write this `void* mem_param` and then local `char* mem = (char*)mem_param`, i.e. in case any math or deref or pass off to other function occurs, but this should be ok.